staging/src/k8s.io/apiserver/pkg/audit/policy/checker.go

```diff
@@ -36,9 +36,29 @@ type Checker interface {
 
 // NewChecker creates a new policy checker.
 func NewChecker(policy *audit.Policy) Checker {
+	for i, rule := range policy.Rules {
+		policy.Rules[i].OmitStages = unionStages(policy.OmitStages, rule.OmitStages)
+	}
 	return &policyChecker{*policy}
 }
 
+func unionStages(s1 []audit.Stage, s2 []audit.Stage) []audit.Stage {
+	m := make(map[audit.Stage]struct{})
+	for _, s := range s1 {
+		m[s] = struct{}{}
+	}
+	for _, s := range s2 {
+		m[s] = struct{}{}
+	}
+	result := make([]audit.Stage, len(m))
+	i := 0
+	for key := range m {
+		result[i] = key
```

Comment:
append, no index i, and `make([]audit.Stage, 0, len(m))`

Reply:
Done.